docs/egress.md

```diff
@@ -0,0 +1,299 @@
+with Antrea v1.2, it can be allocated from an `ExternalIPPool` automatically.
+
+- If `egressIP` is not specified, `externalIPPool` must be specified. An IP will
+  be allocated from the pool by the antrea-controller. The IP will be assigned
+  to a Node selected by the `nodeSelector` of the `externalIPPool` automatically.
+- If both `egressIP` and `externalIPPool` are specified, the IP must be in the
+  range of the pool. Similarly, the IP will be assigned to a Node selected by
+  the `externalIPPool` automatically.
+- If only `egressIP` is specified, Antrea will not manage the assignment of the
+  IP and it must be assigned to an arbitrary interface of one Node manually.
+
+**Starting with Antrea v1.2, high availability is provided automatically when
+the `egressIP` is allocated from an `externalIPPool`**, i.e. when the
+`externalIPPool` is specified. If the Node hosting the `egressIP` fails, another
+Node will be elected (from among the remaining Nodes selected by the
+`nodeSelector` of the `externalIPPool`) as the new egress Node of this Egress.
+It will take over the IP and send layer 2 advertisement (for example, Gratuitous
+ARP for IPv4) to notify the other hosts and routers on the network that the MAC
+address associated with the IP has changed.
+
+**Note**: If more than one Egress applies to a Pod and they specify different
+`egressIP`, the effective egress IP will be selected randomly.
+
+### ExternalIPPool
+
+The `externalIPPool` field specifies the name of the `ExternalIPPool` that the
+`egressIP` should be allocated from. It also determines which Nodes the IP can
+be assigned to. It can be empty, which means users should assign the `egressIP`
+to one Node manually.
+
+## The ExternalIPPool resource
+
+ExternalIPPool defines one or multiple IP ranges that can be used in the
+external network. The IPs in the pool can be allocated to the Egress resources
+as the Egress IPs. A typical ExternalIPPool resource example:
+
+```yaml
+- apiVersion: crd.antrea.io/v1alpha2
+  kind: ExternalIPPool
+  metadata:
+    name: prod-external-ip-pool
+  spec:
+    ipRanges:
+    - start: 10.10.0.2
+      end: 10.10.0.10
+    - cidr: 10.10.1.0/28
+    nodeSelector:
+      matchLabels:
+        network-role: egress-gateway
+```
+
+### IPRanges
+
+The `ipRanges` field contains a list of IP ranges representing the available IPs
+of this IP pool. Each IP range may consist of a `cidr` or a pair of `start` and
+`end` IPs (which are themselves included in the range).
+
+### NodeSelector
+
+The `nodeSelector` field specifies which Nodes the IPs in this pool can be
+assigned to. It's useful when you want to limit egress traffic to certain Nodes.
+The semantics of the selector is the same as those used elsewhere in Kubernetes,
+i.e. both `matchLabels` and `matchExpressions` are supported. It can be empty,
+which means all Nodes can be selected.
+
+## Usage examples
+
+### Configuring high available Egress
```

Comment:
I thought it should be "highly available", but after some researching I am not very certain now. @antoninbas : comments?

Personally I prefer "Egress with HA", or "auto-failover Egress".

Reply:
yes, it should either be "highly-available Egress" or "high-availability Egress"

Reply:
Thanks, updated to "High-Availability", following https://kubernetes.io/docs/tasks/administer-cluster/highly-available-control-plane